lib/nodejs/__tests__/swagger-store.test.js

```diff
@@ -0,0 +1,110 @@
+'use strict';
+
+
+/**
+ * Mocking crypto.createHash('md5').update('...').digest('hex')
+ */
+const mockDigest = jest.fn(() => {
+  return 'DIGEST';
+});
+const mockUpdate = jest.fn(() => {
+  return {
+    digest: mockDigest
+  };
+});
+const mockCreateHash = jest.fn(() => {
+  return {
+    update: mockUpdate
+  };
+});
+const mockCrypto = {
+  createHash: mockCreateHash
+};
+
+jest.mock('crypto',() =>  mockCrypto);
+
+
+function mockSwagger (){
+  this.swaggerObject = '';
+
+  this.validate = () => {
+    this.swaggerObject = 'VALIDATED';
+    return Promise.resolve(this);
+  };
+
+  this.decorate = mockSwagger.prototype.decorate = (x) => {
+    this.swaggerObject += '-' + x();
+    return Promise.resolve(this);
+  };
+}
+
+jest.mock('../swagger', () => ({
+  Swagger:  mockSwagger,
+  decorators: {
+    docExclude: () => 'DOC_EXCLUDE' ,
+    host: () => 'HOST'
+  }
+}));
+
+describe('swagger-store', () => {
+  const mockHexo = {};
+  const swaggerStore = require('../swagger-store')({hexo: mockHexo});
+
+  it('should return swagger.helpers', () => {
+    expect(Object.keys(swaggerStore).length).toBe(5);
+  });
+
+  it('should get digest', () => {
+    const digest = swaggerStore.getDigest('/path/to/file');
+    expect(digest).toEqual('DIGEST');
+    expect(mockCreateHash).toHaveBeenCalled();
+    expect(mockCreateHash).toHaveBeenCalledWith('md5');
+    expect(mockUpdate).toHaveBeenCalled();
+    expect(mockUpdate).toHaveBeenCalledWith('/path/to/file');
+    expect(mockDigest).toHaveBeenCalled();
+    expect(mockDigest).toHaveBeenCalledWith('hex');
+  });
+
+  describe('when called second time with same path', () => {
+    it('should use cache', () => {
+      const digest = swaggerStore.getDigest('/path/to/file');
+      expect(digest).toEqual('DIGEST');
+      expect(mockCreateHash).not.toHaveBeenCalled();
+      expect(mockUpdate).not.toHaveBeenCalled();
+      expect(mockDigest).not.toHaveBeenCalled();
+    });
+  });
+
+  it('should prepare routes', () => {
+    const downloadRoute = swaggerStore.prepareRoute('/path/to/file.yaml');
+    expect(downloadRoute).toBe('DIGEST/file.yaml');
+  });
+
+  it('should set and get route', () => {
+    swaggerStore.setRoute('ROUTE', 'DATA');
+    const routeData = swaggerStore.getRoutes('ROUTE');
+    expect(routeData).toBe('DATA');
+  });
+
+  describe('when no route is passed in setRoute', () => {
+    it('should no do anything', () => {
+      swaggerStore.setRoute();
```

Comment:
@bhaskarmelkani How do you check that "is not doing anything"?

Reply:
Fixed